router/functionHandler.go

```diff
@@ -52,11 +64,12 @@ func (fh *functionHandler) getServiceForFunction() (*url.URL, error) {
 type RetryingRoundTripper struct {
 	maxRetries    int
 	initalTimeout time.Duration
+	transport     http.Transport
 }
 
 func (rrt RetryingRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
 	timeout := rrt.initalTimeout
-	transport := http.DefaultTransport.(*http.Transport)
+	transport := &rrt.transport
```

Comment:
I just verified that setting `DisableKeepAlives` to `true` is working. But seems we are using the `RetryingRoundTripper`, it modifies the timeout field of the transport. If the transport is shared by all the router's proxy requests, it may cause problems?